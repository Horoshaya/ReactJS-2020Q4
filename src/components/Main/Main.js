import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import MovieList from '../MovieList/MovieList';
import { FiltersSortBar } from '../FiltersSortBar/FiltersSortBar';
import { getMovie, sortByGenre } from '../../store/actions/actionCreators';

export const MainContext = React.createContext();

const Main = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMovie());
  }, [dispatch]);

  const sortMovieByGenre = (genre) => {
    dispatch(sortByGenre(genre, movieReducer));
  };

  const movieReducer = useSelector((store) => store.movieReducer);

  return (
    <main>
      <MainContext.Provider value={sortMovieByGenre}>
        <FiltersSortBar />
      </MainContext.Provider>
      <MovieList movieReducer={movieReducer} />
    </main>
  );
};

export default Main;
