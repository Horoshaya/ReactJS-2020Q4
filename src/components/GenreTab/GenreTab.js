import React, { useContext } from 'react';
import styles from './GenreTab.css';
import { MainContext } from '../Main/Main';

const GenreTab = ({ isActive, children }) => {
  const { filterByGenreHandle } = useContext(MainContext);
  return (
    <button
      onClick={() => filterByGenreHandle(children)}
      className={`${styles.button} ${isActive ? styles.active : ''}`}
    >
      {children}
    </button>
  );
};

export default GenreTab;
