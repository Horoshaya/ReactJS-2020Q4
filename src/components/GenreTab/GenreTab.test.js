import React from 'react';
import rerender from 'react-test-renderer';
import GenreTab from './GenreTab';
import { MainContext } from '../Main/Main';

describe('GenreTab', () => {
  test('snapshot sould be created', () => {
    console.log(GenreTab);
    const element = rerender.create(
      <MainContext.Provider value={'test'}>
        <GenreTab />
      </MainContext.Provider>,
    );
    const tree = element.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
