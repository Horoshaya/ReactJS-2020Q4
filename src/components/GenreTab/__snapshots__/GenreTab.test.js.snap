// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreTab snapshot sould be created 1`] = `
<button
  className=""
  onClick={[Function]}
/>
`;

exports[`when GenreTab then snapshot created 1`] = `
<button
  className=""
  onClick={[Function]}
/>
`;
